agent/05_agent_create_cluster.sh

```diff
@@ -102,9 +102,81 @@ function wait_for_cluster_ready() {
   echo "Cluster is ready!"
 }
 
+function wait_for_operator() {
+    subscription="$1"
+    namespace="${2:-}"
+    echo "Waiting for operator ${subscription} to get installed on namespace ${namespace}..."
+
+    for _ in $(seq 1 60); do
+        csv=$(oc -n "${namespace}" get subscription "${subscription}" -o jsonpath='{.status.installedCSV}' || true)
+        if [[ -n "${csv}" ]]; then
+            if [[ "$(oc -n "${namespace}" get csv "${csv}" -o jsonpath='{.status.phase}')" == "Succeeded" ]]; then
+                echo "ClusterServiceVersion (${csv}) is ready"
+                return 0
+            fi
+        fi
+
+        sleep 10
+    done
+
+    echo "Timed out waiting for csv to become ready!"
+    return 1
+}
+
+function install_lso_operator_latest_available() {
+  local version="$(openshift_version ${OCP_DIR})"
+  local operator_version="4.11"
+  local catalog_source_name=redhat-operators-v${operator_version/./-}
+
+  if [[ ( -n "${AGENT_DEPLOY_MCE}" ) && ( ${version} == "4.12" || ${version} == "4.13" ) ]]; then
+    cat > "${mceManifests}/catalog_source_lso_operator.yaml" << EOF
+kind: CatalogSource
+apiVersion: operators.coreos.com/v1alpha1
+metadata:
+  name: ${catalog_source_name}
+  namespace: openshift-marketplace
+spec:
+  displayName: Red Hat Operators v${operator_version}
+  image: registry.redhat.io/redhat/redhat-operator-index:v${operator_version}
+  priority: -100
+  publisher: Red Hat
+  sourceType: grpc
+  updateStrategy:
+    registryPoll:
+      interval: 10m0s
+EOF
+    apply_manifest "$mceManifests/catalog_source_lso_operator.yaml"
```

Comment:
Any specific reason for not having it as a manifest asset directly under `/agent/asset/mce` folder? Otherwise, also for consistency with the other manifests, I'd prefer to have it there.
I guess it could also be added directly as an extra manifest.

Reply:
I think putting it in /agent/asset/mce makes more sense

Reply:
@andfasano @celebdor As per the current implementation, we only need to remove the code from the current file but when we will move the files to the `mce` directory then we need to remove code/files from multiple places when the operator would be available to the 4.12 catalog. Please let me know your thoughts, I am okay with both approaches.

Reply:
@bfournie already started an activity to move the templating via Ansible, so it'd be better if we could stay on that side for the new developments (and keep away from bash). Wouldn't be sufficient just to define the source conditionally in this case?